### ISSUE TITLE
`Range` syntax sugar via `...`

### ISSUE THREAD

**Issue by emil14** [nevalang/neva#719]
## Problem

There's idea than Nevalang is a "simple" language in a way that it doesn't have operators, statments, expressions, etc and just dataflow through components/nodes, kinda like lambda-calculus only have functions. However:

1. That not really true (desugared form of Nevalang is indeed that simple, but with syntax sugar we have const/msg senders, deferred and chained connections, fan-in and fan-out, etc)
2. Main goal is to make programming simpler and having more sugar/forms might help with that (we wouldn't have desugarer at all otherwise). Yes there are more stuff to learn, but maybe that's ok if people capable learning languages like JS, Python or Go that are much more complex than Nevalang.

## Proposal

The idea is to have a special syntax for range over numbers (and maybe other stuff too)

### Before

Here's how range over numbers works now:

```neva
flow Print10Numbers(sig) (sig) {
    Range, Println
    ---
    :sig -> [
        (1 -> range:from),
        (11 -> range:to)
    ]
    range -> println -> :sig
}
```

### After

```neva
flow Print10Numbers(sig) (sig) {
    Println
    ---
    :sig -> 1..11 -> Println -> :sig
}
```

### `Range` V2 (overloading)

As a side-note, `Range` might be overloaded to support more than just numbers.

```neva
foo -> 'a'..'z' -> bar
```

### Action-items

1. Think about syntax `..` vs `...`, `'a'..'z'` vs `'a..z'`
2. Are we sure happy with exclusive version?

**Comment by emil14** [comment]
There are more questions actually

Another thing to think about is that `Range` is not always used with `Lock` or deferred connections (like in example above)

1. Should `..` sugar work without left side `-> ...`?
2. Should `Range` be intended to be used for some cases (at all?) or it's something like `New` or `#bind`

**Comment by emil14** [comment]
Implemented in #731 but only for constant integers. Most likely we'll add support for arbitrary sources
